### PR TITLE
[backend] Switch to opt-in feature flags

### LINKED ISSUE #8764: Switch from "opt-out dev features" to "opt-in feature flags"
## description 
Today, we can define in the [app configuration file](https://github.com/OpenCTI-Platform/opencti/blob/e17b1ed7b3f65435eaeeab4320482b18c7f5aa9f/opencti-platform/opencti-graphql/config/default.json#L9) a list of dev feature flags to disable (opt-out).
This served well in a purely dev context, allowing to easily see in the config file the existing flags, switch them, etc.

However, this approach is counter intuitive in many aspects of our daily operations.
Deploying new platforms with feature flags enabled will be much more clear to everyone, for config, debugging, etc.

## solution
Turn config into : 

```json
  enabled_feature_flags: [
    FEATURE_FLAG_1,
    FEATURE_FLAG_2
  ],
```

And rework the utility code that checks these flags, backend and frontend side.

### PR DESCRIPTION
**Pull request by labo-flg**
### Proposed changes

Switch the opt-out feature flag system into opt-in system.
This is a standard approach, much clearer for everyone.

I also introduce a custom feature flag `*` that basically means "enable all available feature flags".

### Related issues

* closes #8764

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/platformModulesHelper.ts [hunk 1]
```diff
@@ -41,8 +41,9 @@ const isFeatureEnable = (
   id: string,
 ) => {
   const flags = settings.platform_feature_flags ?? [];
-  const feature = flags.find((f) => f.id === id);
-  return feature === undefined || feature.enable === true;
+  // config can target all FF available with special FF id "*"
+  const feature = flags.find((f) => f.id === id || f.id === '*');
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/utils/platformModulesHelper.ts]: Hm, I would have done a big (if flags.includes('*')){ return true; } to be more explicite

**Review by Kedae — APPROVED** [review]

File: opencti-platform/opencti-front/src/utils/platformModulesHelper.ts (hunk 1 above)
Comment (labo-flg) [reply opencti-platform/opencti-front/src/utils/platformModulesHelper.ts]: yep

File: opencti-platform/opencti-front/src/utils/platformModulesHelper.ts [hunk 2]
```diff
@@ -41,8 +41,12 @@ const isFeatureEnable = (
   id: string,
 ) => {
   const flags = settings.platform_feature_flags ?? [];
+  // config can target all FF available with special FF id "*"
+  if (flags.find((f) => f.id === '*' && f.enable)) {
+    return true;
+  }
   const feature = flags.find((f) => f.id === id);
-  return feature === undefined || feature.enable === true;
+  return feature !== undefined && feature.enable === true;
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/utils/platformModulesHelper.ts]:
```suggestion
  return flags.some((f) => f.id === id && f.enable === true);
```

File: .drone.yml [hunk 3]
```diff
@@ -355,7 +355,7 @@ services:
       path: /tmp/e2e-start-platform/node_modules
     environment:
       APP__PORT: 4500
-      APP__DISABLED_DEV_FEATURES: '[]'
+      APP__ENABLED_DEV_FEATURES: '[*]'
```
Comment (labo-flg) [inline .drone.yml]:
```suggestion
      APP__ENABLED_DEV_FEATURES: '["*"]'
```

**Pull request merged**